tests/test_daemon_query.py

```
@@ -0,0 +1,284 @@
+        with patch("tldr.daemon.query_daemon", mock_query_func):
+            # Simulate CLI call: tldr daemon query --json '{"cmd": "semantic",...}'
+            test_args = [
+                "daemon",
+                "query",
+                "--json",
+                json_payload,
+                "--project",
+                str(project_path),
+            ]
+
+            with patch("sys.argv", ["tldr"] + test_args):
```

Ruff (0.14.11)

Consider `["tldr", *test_args]` instead of concatenation

Replace with `["tldr", *test_args]`

(RUF005)